common-extensions/src/main/java/com/twocoders/util/commonextensions/DateExtensions.kt

```diff
@@ -0,0 +1,43 @@
+package com.twocoders.util.commonextensions
+
+import android.content.Context
+import android.text.format.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+
+const val EPOCH_TIME = 0L
+
+val TIME_ZONE_UTC: TimeZone = TimeZone.getTimeZone("UTC")
+
+private const val ONE_DAY = 24L * 60L * 60L * 1000L
+private const val ONE_WEEK = 7L * ONE_DAY
+private const val ONE_MONTH = 31L * ONE_DAY
+private const val ONE_YEAR = 365L * ONE_DAY
+
+fun Date.now() = Date()
+
+fun Date.epochTime() = Date(EPOCH_TIME)
+
+fun Date.dayAgo() = Date(this.time - ONE_DAY)
+
+fun Date.weekAgo() = Date(this.time - ONE_WEEK)
+
+fun Date.monthAgo() = Date(this.time - ONE_MONTH)
+
+fun Date.yearAgo() = Date(this.time - ONE_YEAR)
+
+fun Date.formatDateTime(pattern: String): String = SimpleDateFormat(pattern, Locale.getDefault())
```

Comment:
nie ze by to bolo zle, ale dnes by som uz `Date` na taketo veci vobec nepouzival. `java.time` ma na to funkcie a uz by som ked tak robil extensions na to a Date pouzival iba na minimum veci, nie urcite na formatovanie cez `SimpleDateFormat`

Reply:
Java Time požaduje minimálně API 26 a jeho alternativy jako např. ThreeTen od Jake Whartona potřebuje init, což ještě donedávna šlo na úrovni libky/modulu jen přes ten fake content provider. Zatím mi přijde stále valid na jednoduché věci používat původní Date, i když je to na složitější věci pain.

Reply:
> Java Time požaduje minimálně API 26

Nie celkom... Je to spatne kompatibilne cez desugaring. Uz som to pouzival, je to pododa a lepsie ako davat tam kadejake alternativy 

https://developer.android.com/studio/write/java8-support#library-desugaring

Reply:
oki, tak toto jsem zatím nepoužil. to jak funguje ten desugaring, to ti všechny ty chybějící věci dobalí s sebou do apky nebo jak to funguje?